File: model/request_ad_equipment.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type RequestAdEquipment struct {
	ID            int            `json:"id"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `json:"deleted_at"       swaggertype:"string"`
	Status        string         `json:"status"`
	UserID        int            `json:"user_id"`
	User          User           `json:"user"`
	ExecutorID    *int           `json:"executor_id"`
	Executor      User           `json:"executor"         gorm:"foreignKey:ID;references:ExecutorID"`
	AdEquipmentID int            `json:"ad_equipment_id"`
	AdEquipment   AdEquipment    `json:"ad_equipment"`
	StartLeaseAt  Time           `json:"start_lease_at"`
	EndLeaseAt    Time           `json:"end_lease_at"`
	CountHour     *int           `json:"count_hour"`
	OrderAmount   *int           `json:"order_amount"`
	Description   string         `json:"description"`
	Address       string         `json:"address"`
	Latitude      *float64       `json:"latitude"`
	Longitude     *float64       `json:"longitude"`
	Document      []Document     `json:"-"                 gorm:"many2many:request_ad_equipments_documents"`
	UrlDocument   []string       `json:"url_foto"          gorm:"-"`
}

type FilterRequestAdEquipment struct {
	UserDetail         *bool
	ExecutorDetail     *bool
	AdEquipmentDetail  *bool
	DocumentDetail     *bool
	Unscoped           *bool
	IDs                []int
	AdEquipmentIDs     []int
	AdEquipmentUserIDs []int
	UserIDs            []int
	Limit              *int
	Offset             *int
	Status             *string
}

File: mobile/lib/src/features/requests/data/models/equipment_request/equipment_request_model/request_ad_equipment.dart
import 'package:eqshare_mobile/src/core/data/models/category_models/sub_category.dart';
import 'package:eqshare_mobile/src/core/data/models/city_detail_model/city.dart';
import 'package:eqshare_mobile/src/core/data/models/user_model/user.dart';
import 'package:eqshare_mobile/src/features/ad/data/models/ad_equipment_models/ad_equipment_model/ad_equipment.dart';

import 'package:freezed_annotation/freezed_annotation.dart';



part 'request_ad_equipment.freezed.dart';
part 'request_ad_equipment.g.dart';

@freezed
class RequestAdEquipment with _$RequestAdEquipment {
  const factory RequestAdEquipment({
    required int id,
    @JsonKey(name: 'created_at') DateTime? createdAt,
    @JsonKey(name: 'updated_at') DateTime? updatedAt,
    @JsonKey(name: 'deleted_at') DateTime? deletedAt,
    String? status,
    @JsonKey(name: 'user_id') int? userId,
    User? user,
    @JsonKey(name: 'executor_id') int? executorId,
    User? executor,
    @JsonKey(name: 'ad_equipment_id') int? adEquipmentId,
    @JsonKey(name: 'ad_equipment') AdEquipment? adEquipment,
    @JsonKey(name: 'start_lease_at') String? startLeaseAt,
    @JsonKey(name: 'end_lease_at') String? endLeaseAt,
    @JsonKey(name: 'count_hour') dynamic countHour,
    @JsonKey(name: 'order_amount') dynamic orderAmount,
    @JsonKey(name: 'equipment_sub_category') SubCategory? subCategory,
    String? description,
    String? address,
    double? latitude,
    double? longitude,
    double? price,
    String? title,
    City? city,
 
    @JsonKey(name: 'url_foto') List<String>? urlFoto,
  }) = _RequestAdEquipment;

  factory RequestAdEquipment.fromJson(Map<String, dynamic> json) =>
      _$RequestAdEquipmentFromJson(json);
}
